core/src/test/scala/ai/lum/odinson/lucene/index/TestIncrementalIndex.scala

```diff
@@ -121,4 +122,46 @@ class TestIncrementalIndex extends OdinsonTest with BeforeAndAfterEach {
     index.close()
   }
 
+  it should "incrementally delete Odinson Documents from an open index" in {
+    var index = OdinsonIndex.fromConfig(testConfig)
+
+    // doc w/ 1 sentence & metadata w/ 2 sets of nested fields
+    val pies = getDocument("tp-pies")
+    index.indexOdinsonDoc(pies)
+
+    val odinsonDocId = pies.id
+    index.numDocs() shouldBe 4 // # of lucene docs
+    index.luceneDocIdsFor(odinsonDocId).size shouldBe 4
+
+    // Deleting an Odinson document should delete all of its sentences, as well as its metadata (including any nested fields)
+    index.deleteOdinsonDoc(odinsonDocId)
+
+    index.numDocs() shouldBe 0
+    index.luceneDocIdsFor(odinsonDocId).size shouldBe 0
+    index.close()
```

Comment:
It doesn't matter much, especially in a test, but it's nice to have things close even if exceptions have been thrown.

Reply:
I'll add a context manager to `OdinsonIndex` and override it in `IncrementalOdinsonIndex`.  I hadn't bothered, because I thought the suite was deleting the index dir for each test via the `BeforeAndAfterEach` mixin and `afterEach()` override, but it seems that isn't the case.  I defined something like that for use with the REST API, so I'll just port it over here and make use of it over there after the next release.